### ISSUE TITLE
Prediction using word embeddings

### ISSUE THREAD

**Issue by satishkumar** [mit-nlp/MITIE#128]
Thanks for this great tool.
Can you provide some innards on how the word embeddings work to accomplish the prediction of unseen entities?
For example, I have a sample training dataset where the entities are labelled for every word/phrase in each sentence. The labelling expects a CRF-type algorithm to work efficiently, where nearby words within a context window have semantic relationships. After training with the total_word_feature_extractor.dat with the named entities, will the trained model be able to predict unseen words?
My dataset currently has 26 labels with around 150 samples given to NER trainer.
Here is the partial output with the triggered training:
Part I: train segmenter
words in dictionary: 200000
num features: 271
now do training
C:           20
epsilon:     0.01
num threads: 4
cache size:  5
max iterations: 2000
................
Part II: train segment classifier
now do training
num training samples: 438

How long will it take for the training to complete?

**Comment by davisking** [comment]
How long it takes depends on your data.  For instance, if your labeling is somehow inconsistent then it will take longer to train because the optimizer is trying to do something impossible.  Ultimately, the only way to find out what will happen is to run it and see.  But normally training on a small dataset like that only takes a few minutes.  But it might take longer.

**Comment by satishkumar** [comment]
Thanks for the reply. I had triggered the training and it continues to run for more than 16 hours. 

Do you have some ideas on the other part of my question - i.e. whether the MITIE training is robust for CRF type labelling?. Can MITIE learn semantic relationships between nearby words and use it to identify synonyms(unseen words but that belong to the same entity group as previously learned) in new sentences?

**Comment by davisking** [comment]
Yes, MITIE _**is**_ a CRF.

**Comment by satishkumar** [comment]
Ok. Thanks.
